File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st(nums.begin(),nums.end());
        int ans=0;
        for(auto &x:nums)
        {
            if(st.find(x-1)==st.end())
            {
                int curr=x,len=1;
                while(st.find(curr+1)!=st.end())
                {
                    curr+=1;
                    len++;
                }
                ans=max(ans,len);
            }
        }
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        st = set(nums)
        maxLen = 0
        for num in st:
            if (num - 1) not in st:
                currNum = num
                Len = 1
                while (currNum + 1) in st:
                    Len += 1
                    currNum += 1
                maxLen = max(maxLen, Len)
        return maxLen